packages/core/util.mjs

```diff
@@ -149,6 +151,25 @@ export function curry(func, overload, arity = func.length) {
       const partial = function (...args2) {
         return curried.apply(this, args.concat(args2));
       };
+
+      if (args.length == arity - 1) {
+        // The penultimate arg.. so add some composition magic
+        // TODO - To make this useful, we also need to add stub functions for every pattern method to
+        // do the actual composing
+        for (const r of Pattern.__registered) {
```

Comment:
is __registered already fully populated at the time of executing this? this reminds me of the problem I *tried* to solve with the now extinct Pattern.prototype.bootstrap

Reply:
Not sure, I was hoping this wouldn't be a closure so would always be up to date.

But I'm planning on just returning a dummy pattern with `__compose` as the function instead of doing this.